Pages/ManageIndividualVolunteer.razor

```diff
@@ -0,0 +1,106 @@
+        <input type="text" @bind="@volunteer.Name" @oninput="@((e) => { volunteer.Name = (string)e.Value; })" />
+        <br>
+        <br>
+        <label for="fname">Volunterr surname:</label>
+        <input type="text" @bind="@volunteer.Surname" @oninput="@((e) => { volunteer.Surname = (string)e.Value; })" />
+        <br>
+        <br>
+        <br>
+        <br>
+        <table class="table">
+            <tbody>
+                <tr>
+                    <th>Charity events that this volunteer is working on:</th>
+                    <th>Actions:</th>
+                </tr>
+                @foreach (var charityEvent in GetEventsOfThisVolunteer())
+                {
+                    <tr>
+                        <td>@charityEvent.Name</td>
+                        <td>
+                            <button class="btn btn-primary" @onclick="() => RemoveEventFromList(charityEvent.Id)">Remove</button>
+                        </td>
+                    </tr>
+                }
+                <tr>
+                    <th>Charity events that this volunteer could be a part of:</th>
+                    <th>Actions:</th>
+                </tr>
+                @foreach (var charityEvent in GetEventsNotOfThisVolunteer())
+                {
+                    <tr>
+                        <td>@charityEvent.Name</td>
+                        <td>
+                            <button class="btn btn-primary" @onclick="() => AddEventToVolunteerList(charityEvent.Id)">Add</button>
+                        </td>
+                    </tr>
+                }
+            </tbody>
+        </table>
+        <br>
+        <br>
+        <button class="btn btn-primary" @onclick="UpdateVolunteerData">Update volunteer data</button>
+    }
+    else
+    {
+        <h1>Sorry,but this volunteer was not found</h1>
+    }
+}
+else
+{
+    <label for="fname">Please LogIn</label>
+}
+
+@code
+{
+    [Parameter]
+    public Guid Id { get; set; }
+    private Volunteer volunteer;
+    KarmaContext karmaContext = new KarmaContext();
+
+    private async Task UpdateVolunteerData()
+    {
+        karmaContext.Volunteers.Update(volunteer);
+        karmaContext.SaveChanges();
+        m_uriHelper.NavigateTo("/volunteers");
+        await m_jsRuntime.InvokeVoidAsync("alert", "The volunteer data has been updated");
+    }
+
+    protected override void OnInitialized()
+    {
+        volunteer = karmaContext.Volunteers.Include(p => p.Events).Where(p => p.Id == Id).FirstOrDefault();
+    }
+
+    public IEnumerable<ICharityEvent> GetEventsOfThisVolunteer()
+    {
+        return karmaContext.Volunteers.Include(p => p.Events).Where(p => p.Id == Id).Select(p => p.Events).SelectMany(p => p);
+    }
+
+    public IEnumerable<ICharityEvent> GetEventsNotOfThisVolunteer()
+    {
+        return karmaContext.Events.Include(p => p.Volunteers).Where(p => !p.Volunteers.Contains(volunteer));
+    }
+
+    private void AddEventToVolunteerList(Guid id)
```

Comment:
karmaContext.SaveChanges() after remove delete event.

This will make blazor re-render the page and items will automatically move to the desired list (UX becomes 21st century like)

Reply:
Added karmaContext.SaveChanges() both after remove and add event.